src/libsystemd/sd-journal/journal-file.c

```diff
@@ -4344,6 +4344,11 @@ int journal_file_archive(JournalFile *f, char **ret_previous_path) {
         if (!endswith(f->path, ".journal"))
                 return -EINVAL;
 
+        if (le64toh(f->header->head_entry_seqnum) == 0)
+                /* We may asynchronously copy file later e.g. on btrfs. During copying, if
+                 * journal_directory_vacuum() is called, a broken archive file may be created. */
```

Comment:
hmm, i don't grok this comment? also, it doesn't say anything about that we shouldn't archive empty journal files (which the git commit msg says, and i think this belongs in the comment too)